include/rapidjson/document.h

```diff
@@ -52,6 +53,9 @@ RAPIDJSON_DIAG_OFF(effc++)
 
 RAPIDJSON_NAMESPACE_BEGIN
 
+static const double LOWEST_FLOAT_AS_DOUBLE = static_cast<double>(-3.4028234e38f);
```

Comment:
Global static variable needed to be prevented, because it may be included by multiple implementation files. In other headers I use inline function to return the values instead, if they needed to be named.